### PR TITLE
added new customization 'dayPropGetter'

### PR DESCRIPTION
**Pull request by eschreiner**
Hi there,

I have extended the code with another optional customization function `dayPropGetter(date)` that can be used to customize e.g. the background of days in the month, week, and work week views without the need for custom components, e.g. to render all past dates with a gray background.

Please consider it for merging into master. Let me know if I should update anything (like the function's name) before you accept this pull request.

Best regards,
Erich

### CODE REVIEW

File: src/BackgroundCells.js [hunk 1]
```diff
@@ -56,23 +57,28 @@ class BackgroundCells extends React.Component {
...
+    let { range, cellWrapperComponent: Wrapper, dayPropGetter, date: currentDate } = this.props;
     let { selecting, startIdx, endIdx } = this.state;
 
     return (
       <div className='rbc-row-bg'>
         {range.map((date, index) => {
           let selected =  selecting && index >= startIdx && index <= endIdx;
+          const { className, style: dayStyles } = (dayPropGetter && dayPropGetter(date)) || {};
+          const segmStyles = segStyle(1, range.length)
+          const styles = Object.assign({}, segmStyles, dayStyles)
```
Comment (jquense) [inline src/BackgroundCells.js]: segmentStyles should be assigned in last so they cannot be overridden by the dayPropGetter styles

File: src/DayColumn.js [hunk 2]
```diff
@@ -97,25 +98,30 @@ class DayColumn extends React.Component {
...
+
     return (
       <TimeColumn
         {...props}
         className={cn(
           'rbc-day-slot',
+          className,
           dates.isToday(max) && 'rbc-today'
         )}
+        style={Object.assign({}, slotStyle, style)}
```
Comment (jquense) [inline src/DayColumn.js]: same here `style, slotStyle`

File: src/TimeSlot.js [hunk 3]
```diff
@@ -22,7 +23,7 @@ export default class TimeSlot extends Component {
   }
 
   render() {
-    const { value, slotPropGetter } = this.props;
+    const { value, slotPropGetter, dayPropGetter } = this.props;
```
Comment (jquense) [inline src/TimeSlot.js]: this isn't used here (and shouldn't be)

File: src/TimeSlotGroup.js [hunk 4]
```diff
@@ -25,11 +26,12 @@ export default class TimeSlotGroup extends Component {
   }
 
   renderSlice(slotNumber, content, value) {
-    const { dayWrapperComponent, showLabels, isNow, culture, slotPropGetter } = this.props;
+    const { dayWrapperComponent, showLabels, isNow, culture, slotPropGetter, dayPropGetter } = this.props;
     return (
       <TimeSlot
         key={slotNumber}
         slotPropGetter={slotPropGetter}
+        dayPropGetter={dayPropGetter}
```
Comment (jquense) [inline src/TimeSlotGroup.js]: remove

File: src/TimeGrid.js [hunk 5]
```diff
@@ -201,7 +202,7 @@ export default class TimeGrid extends Component {
   }
 
   renderEvents(range, events, today){
-    let { min, max, endAccessor, startAccessor, components } = this.props;
+    let { min, max, endAccessor, startAccessor, components, dayPropGetter } = this.props;
```
Comment (jquense) [inline src/TimeGrid.js]: doesn't need to be spread out here

File: src/TimeGrid.js [hunk 6]
```diff
@@ -310,9 +315,10 @@ export default class TimeGrid extends Component {
           key={i}
           className={cn(
             'rbc-header',
+            className,
             dates.isToday(date) && 'rbc-today',
           )}
-          style={segStyle(1, this.slots)}
+          style={Object.assign({}, segStyle(1, this.slots), style)}
```
Comment (jquense) [inline src/TimeGrid.js]: order

**Comment by eschreiner** [comment]
Hi Jason,

thank you for the thorough review.

I have updated my code according to your comments. Do I need to do anything additional (besides committing) from my side?

Thanks again & best regards,
Erich

**Comment by eschreiner** [comment]
Hi again,

I have just moved the demo code for the new dayPropGetter to the 'rendering' demo.

- Is there any documentation that I should update / extend accordingly?
- When will you be able to merge these changes and create a new release?

Thank you again & best regards,
Erich

**Comment by eschreiner** [comment]
This new feature provides a solution for #272 .

**Comment by jquense** [comment]
thanks!

**Comment by eschreiner** [comment]
Hi Jason,

thank you for the quick response.

Should we create a new release 0.17.1 for this? Please let me know if you want to do so, then I'll update ChangeLog.md and bump the version in `package.json` accordingly and push these changes to my fork.

Regards,
Erich

**Comment by jquense** [comment]
feel free to PR a change to the Changelog, i will handle bumping the versions

**Comment by kirkjensen18** [comment]
Any idea when you'll publish this to npm?

**Comment by Ronnyrr** [comment]
Any documentation on the usage of this function? Can't find `dayPropGetter` anywhere in the docs?

I only see a reference in this source file, but it's not in my installed files? I just ran `npm i -S react-big-calendar` which installed v0.17.0? 
https://github.com/intljusticemission/react-big-calendar/blob/master/src/Calendar.js#L360 

Do I need to install something else? From a MR or commit?

**Pull request merged**